packages/fx-core/src/component/driver/env/generate.ts

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as dotenv from "dotenv";
+import * as fs from "fs-extra";
+import * as os from "os";
+import { Service } from "typedi";
+
+import { hooks } from "@feathersjs/hooks/lib";
+import { FxError, Result, SystemError, UserError } from "@microsoft/teamsfx-api";
+
+import { getLocalizedString } from "../../../common/localizeUtils";
+import { wrapRun } from "../../utils/common";
+import { logMessageKeys } from "../aad/utility/constants";
+import { DriverContext } from "../interface/commonArgs";
+import { StepDriver } from "../interface/stepDriver";
+import { addStartAndEndTelemetry } from "../middleware/addStartAndEndTelemetry";
+import { InvalidParameterUserError } from "./error/invalidParameterUserError";
+import { UnhandledSystemError } from "./error/unhandledError";
+import { GenerateEnvArgs } from "./interface/generateEnvArgs";
+
+const actionName = "env/generate";
+const helpLink = "https://aka.ms/teamsfx-actions/env-generate";
+
+@Service(actionName) // DO NOT MODIFY the service name
+export class GenerateEnvDriver implements StepDriver {
+  @hooks([addStartAndEndTelemetry(actionName, actionName)])
+  public async run(
+    args: GenerateEnvArgs,
+    context: DriverContext
+  ): Promise<Result<Map<string, string>, FxError>> {
+    return wrapRun(() => this.handler(args, context));
+  }
+
+  private async handler(
+    args: GenerateEnvArgs,
+    context: DriverContext
+  ): Promise<Map<string, string>> {
+    try {
+      this.validateArgs(args);
+
+      if (args.target) {
+        await fs.ensureFile(args.target);
+        const envs = dotenv.parse(await fs.readFile(args.target));
+        const content = Object.entries({ ...envs, ...args.envs })
+          .map(([key, value]) => `${key}=${value}`)
+          .join(os.EOL);
+        await fs.writeFile(args.target, content);
```

Comment:
If `.env` file contains comments, will they be still existing?

Reply:
No. To improve it later if needed.